include/umf/memory_provider.h

```diff
@@ -156,6 +156,34 @@ umfMemoryProviderGetName(umf_memory_provider_handle_t hProvider);
 ///
 UMF_EXPORT umf_memory_provider_handle_t umfGetLastFailedMemoryProvider(void);
 
+///
+/// @brief Splits a coarse grain allocation into 2 adjacent allocations that
+///        can be managed (freed) separately.
+/// @param hProvider handle to the memory provider
+/// @param ptr pointer to the beginning of the allocation
+/// @param totalSize total size of the allocation to be split
```

Comment:
I see that in split/merge methods we don't want to have redundant arguments, so why we need "totalSize"? Shouldn't the totalSize be just the size of the alloc pointed by ptr? What if totalSize would be less/more than allocation?

Reply:
Here are my thoughts about this question:

1. We should not just copy jemalloc extent API because for me it looks jemalloc-specific. And it looks like we are not doing that.
2. Regarding `totalSize`. For me, it looks similar to `umfMemoryProviderFree(..., void *ptr, size_t size)` API that also requires passing the `size` of allocation pointed by `ptr`. But not all memory provider implementations need/use it: e.g. mmap implementation uses `size`, but L0 ignores it.
3. For me, bullet 2 means that UMF requires that the pool manager remember the size of each particular coarse-grain allocation but memory providers are not required to remember the size of each coarse-grain allocation. Since we already require to pass the `size` to `umfMemoryProviderFree` function what is the difference here? 
4. Regarding what if `totalSize != actual total size`, the same question is applicable for `umfMemoryProviderFree`. On our side we can do (at least debug) verification using the tracking memory provider - there is such information in the tracker.
5. Another question regarding input parameter validation, should the `firstSize` be a multiple of the page size?

Reply:
ok I agree with bullet 3

Reply:
Regarding 5: I think it is implied because I'm saying that the pointer needs to point to the beginning of the allocation and all allocations need to be page aligned.